pkg/apiserver/protocalshield.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+package apiserver
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// protocolShieldSerializer filters out media types that the apiserver doesn't accept.
+type protocolShieldSerializer struct {
+	*serializer.CodecFactory
+	accepts []runtime.SerializerInfo
+}
+
+func newProtocolShieldSerializer(codecs *serializer.CodecFactory) *protocolShieldSerializer {
+	if codecs == nil {
+		return nil
+	}
+	pss := &protocolShieldSerializer{
+		CodecFactory: codecs,
+		accepts:      []runtime.SerializerInfo{},
+	}
+	for _, info := range codecs.SupportedMediaTypes() {
+		if info.MediaType == runtime.ContentTypeProtobuf {
+			continue
```

Comment:
Should we only allow `application/json` here as thats pretty much the only thing we support?

Reply:
We could but I guess the end result is the same as this. I'm fine to keep this if you are, but if you want me to change it I will

Reply:
I'm inclined to agree that it feels better to select the media type(s) that we support rather than filtering out known ones that we don't.

Reply:
Updated